app.js

```diff
@@ -1,53 +1,96 @@
-const FileTextEditor = require("./textEditorService");
-const parsers = require("./parsers");
-
-function createFile(action) {
-  const fileEditor = new FileTextEditor(action.params.PATH);
-  return fileEditor.createFile({
-    content: action.params.CONTENT,
-  });
-}
+const fs = require("fs/promises");
+const { bootstrap } = require("kaholo-plugin-library");
+const { tryCreateRegexFromString, parsePath, pathExists } = require("./helpers");
+
+async function createFile({
+  PATH: path,
+  CONTENT: content,
+}) {
+  const parsedPath = parsePath(path);
+
+  try {
+    await fs.writeFile(parsedPath, content);
+  } catch (error) {
+    throw new Error(`Error creating file ${path}: ${error.message || JSON.stringify(error)}`);
+  }
 
-function createVaultFile(action) {
-  return createFile(action);
+  return `Created file ${path}.`;
 }
 
-function appendToFile(action) {
-  const fileEditor = new FileTextEditor(action.params.PATH);
-  return fileEditor.appendToFile({
-    content: action.params.CONTENT,
-    createIfNeeded: !parsers.boolean(action.params.dontCreate),
-    returnContent: parsers.boolean(action.params.return),
-  });
+async function appendToFile({
+  PATH: path,
+  CONTENT: content,
+  return: returnContent,
+  dontCreate,
+}) {
+  const parsedPath = parsePath(path);
+  if (!await pathExists(parsedPath)) {
+    if (!dontCreate) {
```

Comment:
Please flatten the conditions.
or
If there's a situation where one of the condition's branches would fail or return early then put that into the conditional sttement. Try to structure the code in such way that there's only one happy path - at the bottom of the function.
It's ok to nest conditions if it's an unhappy path terminating the function execution.